x/crosschain/keeper/msg_server_update_tss.go

```diff
@@ -70,3 +72,9 @@ func (k msgServer) UpdateTssAddress(
 
 	return &types.MsgUpdateTssAddressResponse{}, nil
 }
+
+// GetChainsSupportingMigration returns the chains that support migration.
+func (k *Keeper) GetChainsSupportingMigration(ctx sdk.Context) []chains.Chain {
+	return append(k.zetaObserverKeeper.GetSupportedForeignChainsByConsensus(ctx, chains.Consensus_ethereum),
```

Comment:
I would prefer a functional programming approach.

We implement a stateless function `FilterChainsSupportingTSSMigration(chainList) -> chainList`

GetChainsSupportingTSSMigration can call `GetSupportedForeignChains` and then apply this method to it.

Unit tests are simpler to develop and maintain.

Reply:
And use can also combine these like so:
```go
type ChainFilter func (c Chain) bool

func SupportsTSS(c Chain) bool {
 // ...
}

func (Keeper) GetSupportedForeignChains(filters ...ChainFilter) []Chain {
   // ...
}

// ....

chainsWithTSSMIgration := keeper.GetSupportedForeignChains(SupportsTSS)

```

Reply:
Thanks, both good ideas. Ill implement them

Reply:
e32c2a87997922c99ad8a9e7709df90d10050b69

Reply:
e32c2a87997922c99ad8a9e7709df90d10050b69